### ISSUE TITLE
Configure option to use standard JSON.stringify instead of Angular's own implementation on HttpClient post

### ISSUE THREAD

**Issue by rcarduner** [angular/angular#38117]
# 🚀 feature request

### Relevant Package
This feature request is for @angular/common/http - HttpClient

### Description
Would like the ability to use the standard Javascript JSON.stringify() method when the body is serialized, rather than Angular's "simplified" version.  Alternately, Angular's version should call toJSON() on any objects in the graph that implement it.

The specific issue I'm having is that I have a parent object and a related child object.  The assigned child object can be changed on the client, and while I am in the client, I want the child object to be the fully-fledged child.  However, when I post the parent object back to the server to save it, I only want the *id* (and a few other properties) of the child object sent, not it's full graph.  An obvious way to do this would be to override the toJSON() method on the child object, but Angular apparently ignores it when serializing.

### Describe the solution you'd like
Option to use the standard JSON.stringify() method, or Angular respects any toJSON() overrides.

### Describe alternatives you've considered
Looked into related issue https://github.com/angular/angular/issues/32117, which suggests using an interceptor.  This is overkill for us.  The main issue is that we have been sending extra unnecessary data to our server on the save call, so it hasn't been a high priority, however due to recent server-side changes it's now a problem having the extra object.  Will null it out on the server (keeping only the ID) rather than cloning the object (minus the extra properties) on the client.

**Comment by petebacondarwin** [comment]
I am a bit confused here. As far as I can tell we already use the standard JSON.stringify to serialise objects when making requests: https://github.com/angular/angular/blob/d1ea1f4c7f3358b730b0d94e65b00bc28cae279c/packages/common/http/src/request.ts#L257

Am I missing something here?

**Comment by rcarduner** [comment]
Apparently I am the one that is confused.  When I could not see my toJSON() methods being called and originally researched this, I was led to believe from one of the comments in the issue I referenced that this was not the case:
![image](https://user-images.githubusercontent.com/15037926/87819778-d4149b00-c83a-11ea-9edd-2d6c6d358c6c.png)

However, it does appear to work now; I'm not sure what was going on previously.  My apologies, and thank you for the quick response.  Closing.
